src/main/java/org/broadinstitute/hellbender/tools/exome/SegmentUtils.java

```diff
@@ -13,292 +15,452 @@
+        return targetSegmentsFixed;
     }
-    
-    
+
     /**
-     *
-     * @param interval some genomic region
-     * @param targets collection of target genomic regions
-     * @param snps collection of SNP sites represented as length-1 intervals
-     *
-     * @return the smallest interval that contains the entirety (start to end) of every interval in {@code targets} and
-     * {@code snps} that overlaps {@code interval} at all.  If {@code interval} overlaps no targets or SNPs, return a length-1 interval
-     * at the start of {@code interval}.
-     *
-     * For example, if {@code interval} is 1:100-200 and the only target or snp it overlaps is 1:60-180, then the result
-     * is 1:60-180.
-     *
-     * Note that if {@code targets} contains overlapping targets we can get an inconsistent result where
-     * parsimoniousInterval returns an interval that is not parsimonious by its own definition.  For example, if {@code targets}
-     * contains 1:40-90 and 1:80-110 and {@code interval} is 1:100-200, then parsimoniousInterval yields 1:80-110, which
-     * falls in the middle of a target.  Since sites in {@code snps} are length-1, there is no such concern for them.
+     * Given an interval and collections of targets and SNPs, returns a trimmed interval produced by removing the empty
+     * portions at the start and the end of the original interval that do not overlap the targets and SNPs that overlap
+     * with the original interval.  If this procedure would remove the entire interval, the original interval is
+     * returned instead.
      */
-    public static SimpleInterval parsimoniousInterval(final Locatable interval, final TargetCollection<? extends Locatable> targets,
-                                                                     final TargetCollection<? extends Locatable> snps) {
-        Utils.nonNull(interval, "Interval may not be null");
-        Utils.nonNull(targets, "Targets may not be null");
-        Utils.nonNull(snps, "SNPs may not be null");
+    public static SimpleInterval trimInterval(final Locatable interval,
+                                              final TargetCollection<? extends Locatable> targets,
+                                              final TargetCollection<? extends Locatable> snps) {
+        Utils.nonNull(interval, "The interval cannot be null.");
+        Utils.nonNull(targets, "The collection of targets cannot be null.");
+        Utils.nonNull(snps, "The collection of SNPs cannot be null.");
 
         final IndexRange targetRange = targets.indexRange(interval);
         final IndexRange snpRange = snps.indexRange(interval);
 
-        final int targetsInInterval = targetRange.size();
-        final int snpsInInterval = snpRange.size();
+        final int numTargetsInInterval = targetRange.size();
+        final int numSNPsInInterval = snpRange.size();
 
-        int start;
-        int end;
+        int start = interval.getStart();
```

Comment:
For some of the comments below, these variables would instead be `SimpleInterval start = new SimpleInterval(interval.getContig(), interval.getStart(), interval.getStart())` and likewise for `end`.